src/agent/src/rpc.rs

```diff
@@ -1343,11 +1348,16 @@ impl protocols::health_ttrpc::Health for HealthService {
     }
 }
 
-fn get_memory_info(block_size: bool, hotplug: bool) -> Result<(u64, bool)> {
+fn get_memory_info(
+    block_size: bool,
+    hotplug: bool,
+    block_size_path: &str,
+    hotplug_probe_path: &str,
+) -> Result<(u64, bool)> {
     let mut size: u64 = 0;
     let mut plug: bool = false;
     if block_size {
-        match fs::read_to_string(SYSFS_MEMORY_BLOCK_SIZE_PATH) {
+        match fs::read_to_string(block_size_path) {
             Ok(v) => {
                 if v.is_empty() {
                     info!(sl!(), "string in empty???");
```

Comment:
This looks like a typo so you could consider creating an initial commit which fixes this to something like the following (we don't need punctuation in error messages so periods, exclamation marks and question marks can -- and ideally should -- be removed imho):

```suggestion
                    info!(sl!(), "string is empty");
```

Reply:
One other point about the existing function: along with returning the errors, it also makes a number of log calls. However, two different log levels are being used `info!()` and `warn!()`. For consistency, they should all be `warn!()` log calls or possibly all `debug!()` calls since the scenarios are handled by returning errors.